### ISSUE TITLE
Visual Studio Code Snippets Collection

### ISSUE THREAD

**Issue by pmacMaps** [Esri/esri-leaflet#1203]
I was reminded today that there is an [ArcGIS API for JavaScript Snippets](https://marketplace.visualstudio.com/items?itemName=Esri.arcgis-jsapi-snippets) extension for VS Code.  The project is maintained on [GitHub](https://github.com/Esri/arcgis-js-vscode-snippets).

It got me thinking that it might be cool to create a spin-off repo to create a VS Code Snippets extension for Esri-Leaflet.

I don't know what's involved in this type of project, but I think having it community-focused would be cool.  I would love to hear other users' thoughts.

**Comment by hhkaos** [comment]
Hi @Groot-2001, the repo of that VS Code extension is here: https://github.com/Esri/arcgis-js-vscode-snippets

I assume the goal here would be to do something similar but using Esri Leaflet, right? 

I guess snippets should help developers to easily:
* Load Esri basemap layers with Leaflet
* Load feature layers
* Load the geocoding search control
* etc.

I would probably grab ideas [from the Sample Code doc](https://developers.arcgis.com/esri-leaflet/samples/).

Should I be a new repo? If you want, I can request a new empty repo that we can use.

**Comment by Groot-2001** [comment]
@hhkaos 

Yes, creating a new repository would be the way to go for building a VS Code extension using Esri Leaflet. The snippets you mentioned can indeed help developers easily incorporate functionalities like loading Esri basemap layers, feature layers, geocoding search, etc. Feel free to request a new empty repository that you can use for this project.

**Comment by hhkaos** [comment]
@Groot-2001 I have been checking with the team and we think for now we would prefer you to host that repo, and we will reference it from this repo and contribute to it.  Would that work?

**Comment by hhkaos** [comment]
hi @Groot-2001, let me know if you need any help with this 🙂

**Comment by patrickarlt** [comment]
Closing as stale
